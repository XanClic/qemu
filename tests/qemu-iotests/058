#!/bin/bash
#
# bochs format input validation tests
#
# Copyright (C) 2013 Red Hat, Inc.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#

# creator
owner=kwolf@redhat.com

seq=`basename $0`
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!

_cleanup()
{
	_cleanup_test_img
}
trap "_cleanup; exit \$status" 0 1 2 3 15

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

_supported_fmt bochs
_supported_proto generic
_supported_os Linux

catalog_size_offset=$((0x48))

echo
echo "== Read from a valid image =="
cp ./058-data/template.bochs "$TEST_IMG"
{ $QEMU_IO -c "read 0 512" $TEST_IMG; } 2>&1 | _filter_qemu_io | _filter_testdir

echo
echo "== Negative catalog size =="
cp ./058-data/template.bochs "$TEST_IMG"
poke_file "$TEST_IMG" "$catalog_size_offset" "\xff\xff\xff\xff"
{ $QEMU_IO -c "read 0 512" $TEST_IMG; } 2>&1 | _filter_qemu_io | _filter_testdir

# success, all done
echo "*** done"
rm -f $seq.full
status=0
