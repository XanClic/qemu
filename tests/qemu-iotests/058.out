QA output created by 058

== Read from a valid image ==
read 512/512 bytes at offset 0
512 bytes, X ops; XX:XX:XX.X (XXX YYY/sec and XXX ops/sec)

== Negative catalog size ==
Catalog size is too large
qemu-io: can't open device TEST_DIR/t.bochs
no file open, try 'help open'

== Overflow for catalog size * sizeof(uint32_t) ==
Catalog size is too large
qemu-io: can't open device TEST_DIR/t.bochs
no file open, try 'help open'

== Too small catalog bitmap for image size ==
Catalog size is too small for this disk size
qemu-io: can't open device TEST_DIR/t.bochs
no file open, try 'help open'
*** done
